### PR TITLE
Add spec for persist module

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose

Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/3434

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+2. Insert Types
+    These are records used to insert data in the data source. It is the same as the entity type.
+    ```ballerina
+    type WorkspaceInsert Workspace;
+    ```
+
+3. Update Types
+    These are records used to update data in the data source. These are entity types without the identifier fields.
+    ```ballerina
+    type WorkspaceUpdate record {|
```
Comment (sahanHe) [inline docs/spec/spec.md:237]: shouldn't these be nullable types, as in [here](https://github.com/ballerina-platform/module-ballerina-persist/blob/005e6fafd3a21fec030880da856a604728eac401/ballerina/tests/generated_types.bal#L58)

Comment (niveathika) [reply docs/spec/spec.md:237]: What you mean is its optional field? `int field?;` ?

Comment (sahanHe) [reply docs/spec/spec.md:237]: Sorry it should be `optional field`

File: docs/spec/spec.md [hunk 2]
```diff
@@ -0,0 +1,404 @@
+# Specification: Ballerina SQL Library
```
Comment (daneshk) [inline docs/spec/spec.md:1]:
```suggestion
# Specification: Ballerina Persist Library
```

File: docs/spec/spec.md [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+
+_Owners_: @daneshk @niveathika @kaneeldias
+_Reviewers_: @daneshk  
+_Created_: 2022/01/25   
+_Updated_: 2022/01/25  
+_Edition_: Swan Lake  
+
+## Introduction
+
+This is the specification for the Ballerina persist standard library of [Ballerina language](https://ballerina.io/), which provides functionality to define and store data.
```
Comment (daneshk) [inline docs/spec/spec.md:11]:
```suggestion
This is the specification for the Ballerina persist standard library of [Ballerina language](https://ballerina.io/), which provides functionality to store and query data easily.  
```

File: docs/spec/spec.md [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+    2.3. [Relationship Definition](#23-relationship-definition)    
+        2.3.1. [One-to-one (1-1)](#231-one-to-one-1-1)  
+        2.3.2. [One-to-Many (1-n)](#232-one-to-many-1-n)  
+3. [Derived Entity Types and Persist Clients](#3-derived-entity-types-and-persist-clients)  
+    3.1. [Derived Entity Types](#31-derived-entity-types)  
+    3.2. [Persist Clients](#32-persist-clients)
+
+## 1. Overview  
+
+The `persist` standard library creates a programming model to store and query data easily. This uses the statically typed nature of Ballerina to provide a convenient way to define and store data model of an application across various providers such as RDBMS, NoSQL, and others. At this initial release `persist` library only supports MySQL data provider.
```
Comment (daneshk) [inline docs/spec/spec.md:36]:
```suggestion
The `persist` standard library creates a programming model to store and query data easily. This uses the statically typed nature of Ballerina to provide a convenient way to define and store the data model of an application across various providers such as RDBMS, NoSQL, and others. At this initial release `persist` library only supports MySQL data provider.
```

File: docs/spec/spec.md [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+    };
+
+    public function close() returns persist:Error? {
+    }
+}
+```
+
+The conventions used in deriving the Persist client are as follows:
+1.  The Client name is derived from the name of the file. Example: `rainier.bal` will generate a client named `RainierClient`(<First Letter Capitalized File Name>Client).
+2. The client should be of `persist:AbstractPersistClient` type.
```
Comment (daneshk) [inline docs/spec/spec.md:279]:
```suggestion
1. The Client name is derived from the name of the file. Example: `rainier.bal` will generate a client named `RainierClient`(<First Letter Capitalized File Name>Client).
2. The client should be of `persist:AbstractPersistClient` type.
```

**Review by daneshk — APPROVED** [review]
Add a few comments. Overall LGTM

File: docs/spec/spec.md [hunk 6]
```diff
@@ -8,7 +8,7 @@ _Edition_: Swan Lake
 
 ## Introduction
 
-This is the specification for the Ballerina persist standard library of [Ballerina language](https://ballerina.io/), which provides functionality to store and query data easily.  
+This is the specification for the Ballerina persist standard library of [Ballerina language](https://ballerina.io/), which provides functionality to store and query data easily.
```
Comment (praneesha) [inline docs/spec/spec.md:11]:
```suggestion
This is the specification for the `persist` standard library of the [Ballerina language](https://ballerina.io/), which provides functionality to store and query data easily.
```

File: docs/spec/spec.md [hunk 7]
```diff
@@ -8,7 +8,7 @@ _Edition_: Swan Lake
...
 
 The persist library specification has evolved and may continue to evolve in the future. The released versions of the specification can be found under the relevant GitHub tag.
```
Comment (praneesha) [inline docs/spec/spec.md:13]:
```suggestion
The `persist` library specification has evolved and may continue to evolve in the future. The released versions of the specification can be found under the relevant GitHub tag. 
```
Please check for the backticks in all occurrences of the module name.

File: docs/spec/spec.md [hunk 8]
```diff
@@ -29,9 +29,9 @@ The conforming implementation of the specification is released and included in t
         2.3.2. [One-to-Many (1-n)](#232-one-to-many-1-n)  
 3. [Derived Entity Types and Persist Clients](#3-derived-entity-types-and-persist-clients)  
     3.1. [Derived Entity Types](#31-derived-entity-types)  
-    3.2. [Persist Clients](#32-persist-clients)
+    3.2. [Persist Clients](#32-persist-clients)  
 
-## 1. Overview  
+## 1. Overview
 
 The `persist` standard library creates a programming model to store and query data easily. This uses the statically typed nature of Ballerina to provide a convenient way to define and store the data model of an application across various providers such as RDBMS, NoSQL, and others. At this initial release `persist` library only supports MySQL data provider.
```
Comment (praneesha) [inline docs/spec/spec.md:36]:
```suggestion
The `persist` standard library creates a programming model to store and query data easily. This uses the statically typed nature of Ballerina to provide a convenient way to define and store the data model of an application across various providers such as RDBMS, NoSQL, and others. At this initial release, the `persist` library only supports the MySQL data provider.
```

File: docs/spec/spec.md (hunk 6 above)
Comment (niveathika) [reply docs/spec/spec.md]: Fixed in [e4dc132](https://github.com/ballerina-platform/module-ballerina-persist/pull/109/commits/e4dc1324152adf53e52a1ca4dea90ffacb44ae7a)

File: docs/spec/spec.md (hunk 8 above)
Comment (niveathika) [reply docs/spec/spec.md]: Fixed in [e4dc132](https://github.com/ballerina-platform/module-ballerina-persist/pull/109/commits/e4dc1324152adf53e52a1ca4dea90ffacb44ae7a)

**Review by praneesha — APPROVED** [review]

**Pull request merged**